erpnext/projects/doctype/timesheet/timesheet.py

```diff
@@ -290,3 +292,31 @@ def get_activity_cost(employee=None, activity_type=None):
 			["costing_rate", "billing_rate"], as_dict=True)
 
 	return rate[0] if rate else {}
+	
+@frappe.whitelist()
+def get_events(start, end, filters=None):
+	"""Returns events for Gantt / Calendar view rendering.
+	:param start: Start date-time.
+	:param end: End date-time.
+	:param filters: Filters (JSON).
+	"""
+	from frappe.desk.calendar import get_event_conditions
+	filters = json.loads(filters)
+
+	data = []
+	if "employee" in filters:
+		employee = filters["employee"]
+		del filters["employee"]
+		filters = json.dumps(filters)
+		conditions = get_event_conditions("Timesheet", filters)
+		data = frappe.db.sql("""select name, parent, from_time, hours, activity_type, project, docstatus from `tabTimesheet Detail` where (ifnull(from_time, '0000-00-00')!= '0000-00-00') \
+						and (from_time between %(start)s and %(end)s) \
+						and parent in (select name from `tabTimesheet` where employee = %(employee)s)	{conditions}""".format(conditions=conditions), {
+					"start": start,
+					"end": end,
+					"employee": employee
+				}, as_dict=True, update={"allDay": 0})
+		for r in data:
+			tmp = {"end_time": r.from_time + timedelta(hours=r.hours)}
+			r.update(tmp)
+	return data
```

Comment:
Replace above method with below code

@frappe.whitelist()
def get_events(start, end, filters=None):
	"""Returns events for Gantt / Calendar view rendering.
	:param start: Start date-time.
	:param end: End date-time.
	:param filters: Filters (JSON).
	"""
	filters = json.loads(filters)

	conditions = get_conditions(filters)
	return frappe.db.sql("""select `tabTimesheet Detail`.name as name, `tabTimesheet Detail`.parent as parent,
		from_time, hours, activity_type, project, to_time as end_time from `tabTimesheet Detail`, 
		`tabTimesheet` where `tabTimesheet Detail`.parent = `tabTimesheet`.name and 
		(from_time between %(start)s and %(end)s) {conditions}""".format(conditions=conditions),
		{
			"start": start,
			"end": end
		}, as_dict=True, update={"allDay": 0})

def get_conditions(filters):
	conditions = []
	abbr = {'employee': 'tabTimesheet', 'project': 'tabTimesheet Detail'}
	for key in filters:
		if filters.get(key):
			conditions.append("`%s`.%s = '%s'"%(abbr.get(key), key, filters.get(key)))

	return " and {}".format(" and ".join(conditions)) if conditions else ""